lib/util/upload_utils.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:http/http.dart' as http;
+
+typedef ProgressCallback = void Function(double progress);
+typedef OnDoneCallback = void Function();
+typedef OnErrorCallback = void Function(dynamic error);
+
+class UploadUtils {
+  
+  static Future<void> uploadFiles(
+      String url, 
+      List<File> files, 
+      {
+        ProgressCallback? progressCallback,
+        OnDoneCallback? onDone,
+        OnErrorCallback? onError,
+      }
+    ) async {
+    final request = MultipartRequest(
+      'POST',
+      Uri.parse(url),  
+      onProgress: (int bytes, int total) {
+        final progress = bytes / total;
+        progressCallback?.call(progress);
+      }
+  );
+    final multipartFiles = <http.MultipartFile>[];
+  
+    for (var file in files) {
+      final multipartFile = await http.MultipartFile.fromPath('files', file.path);
+      multipartFiles.add(multipartFile);
+    }
+
+    request.files.addAll(multipartFiles);
+
+    try {
+      final response = await request.send();
+      
+      if (response.statusCode >= 200 && response.statusCode <= 300) {
+        onDone?.call();
+      } else {
+        throw Exception('Failed to upload file');
+      }
+    } catch (error) {
+      onError?.call(error);
+    }
+    
+  }
+}
+
+
+class MultipartRequest extends http.MultipartRequest {
```

Comment:
This is cool. What's the requirement on the server side to accept files?

Reply:
Requirements.
- API method is expected to be POST and accepts content type multipart/form-data

Side note
To access file server code can use either `request.files` or `request.data['files']`